vernac/lemmas.ml

```diff
@@ -299,20 +299,14 @@ let save_remaining_recthms env sigma ~poly ~scope norm univs body opaq i
         let kn = declare_constant name ~local (DefinitionEntry const, k) in
         (ConstRef kn,impargs)
 
-let initialize_named_context_for_proof () =
-  let sign = Global.named_context () in
-  List.fold_right
-    (fun d signv ->
-      let id = NamedDecl.get_id d in
-      let d = if variable_opacity id then NamedDecl.drop_body d else d in
-      Environ.push_named_context_val d signv) sign Environ.empty_named_context_val
-
 (* Starting a goal *)
 let start_lemma ~name ~poly
     ?(udecl=UState.default_univ_decl)
-    ?(sign=initialize_named_context_for_proof())
     ?(info=Info.make ())
     sigma c =
+  (* Should we do any cleaning of sign here? Previous code claimed to
+     remove "VM code" and bodies of section variables *)
+  let sign = Global.named_context_val () in
```

Comment:
What breaks if we do the drop_body stuff unconditionally here? (instead of the current change which never does it)

Reply:
That seems indeed the right way to proceed for this PR in the short term, I'll indeed postpone the change but I'd like to open an issue about `Let` before.

This PR was motivated as an attempt to remove the use of the "variable opacity" table, but indeed that should wait.

I got a feeling tho that we may be close to actually refining the type of environments used for proofs [vs the "global" one], but still a lot of homework to do in order to fully understand the picture.

Reply:
@ppedrot I'd like to open an issue about `Let`, and I'd like to summarize the different viewpoints; what's your opinion on them and the related discussion in this PR?